website/docs/plugin/framework/functions/documentation.mdx

```diff
@@ -0,0 +1,121 @@
+
+If there are no description formatting differences, set only one of `Description` or `MarkdownDescription`. When Terraform has not sent a preference for the description formatting, the framework will return `MarkdownDescription` if both are defined.
+
+In this example, the function parameters set name and description documentation:
+
+```go
+func (f *CidrContainsIpFunction) Definition(ctx context.Context, req function.DefinitionRequest, resp *function.DefinitionResponse) {
+    resp.Definition = function.Definition{
+        // ... other fields ...
+        Parameters: []function.Parameter{
+            function.StringParameter{
+                Name:        "cidr",
+                Description: "RFC4632 CIDR to check whether it contains the given IP address",
+            },
+            function.StringParameter{
+                Name:        "ip",
+                Description: "IP address to check whether its contained in the RFC4632 CIDR",
+            },
+        },
+    }
+}
+```
+
+## Registry-Based Documentation
+
+Add Markdown documentation files in conventional provider codebase locations before [publishing](/terraform/registry/providers/publishing) to the [Terraform Registry](https://registry.terraform.io). The documentation is displayed and discoverable on the web. These files can be manually created or automatically generated using tooling such as [`terraform-plugin-docs`](https://github.com/hashicorp/terraform-plugin-docs).
+
+The [Registry provider documentation](/terraform/registry/providers/docs) covers the overall requirements, conventional file layout details, and how to enable additional features such as sub-categories for the navigation sidebar. Function documentation for most providers is expected under the `docs/functions/` directory with a file named after the function and with the extension `.md`. Older providers using the legacy file layout use `website/docs/functions/` and `.html.md`.
+
+Functions are conventionally documented with the following:
+
+* Description
+* Example Usage
+* Signature
+* Arguments
+
+In this example, a `docs/functions/contains_ip.md` file (either manually or automatically created) will be displayed in the Terraform Registry after provider publishing:
+
+``````markdown
+---
+page_title: contains_ip Function - terraform-provider-cidr
+description: |-
+  Returns a boolean whether a RFC4632 CIDR contains an IP address.
+---
+
+# Function: contains_ip
+
+Returns a boolean whether a RFC4632 CIDR contains an IP address.
+
+## Example Usage
+
+```terraform
+# result: true
+provider::cidr::contains_ip("10.0.0.0/8", "10.0.0.1")
+```
+
+## Signature
+
+```text
+contains_ip(cidr string, ip string) bool
+```
+
+## Arguments
+
+1. `cidr` (String) RFC4632 CIDR to check whether it contains the given IP address.
+2. `ip` (String) IP address to check whether its contained in the RFC4632 CIDR.
+``````
```

Comment:
Will try to replicate and followup with the web platform team, if necessary, and report back here.

Reply:
Wouldn't you know... this was an easy fix/workaround: https://github.com/hashicorp/terraform-plugin-framework/pull/889/commits/9ebe29046e02d2a3053daf5e68cbf1da787f71ae

My hunch is that refractor's Markdown renderer didn't like the `---` frontmatter and this had nothing to do with the extra long fences. Since the syntax highlighting isn't really important for this snippet, plaintext is certainly fine! The web folks may followup to see if they can do anything to catch/fix this edge case, but I think we're good now. 👍

Reply:
Awesome, nice find!